lib/default-theme/Layout.vue

```diff
@@ -173,6 +181,35 @@ function updateMetaTags (meta, current) {
     })
   }
 }
+
+function setActiveHash () {
+  const anchors = gatherHeaderAnchors()
+
+  const scrollTop = Math.max(window.pageYOffset, document.documentElement.scrollTop, document.body.scrollTop)
+
+  for (let i = 0; i < anchors.length; i++) {
+    const anchor = anchors[i]
+    const nextAnchor = anchors[i + 1]
+
+    const isActive = i === 0 && scrollTop === 0 ||
+      (scrollTop >= anchor.parentElement.offsetTop + 10 &&
+        (typeof nextAnchor === 'undefined' || scrollTop < nextAnchor.parentElement.offsetTop - 10))
+
+    if (isActive && window.location.hash !== anchor.hash) {
+      window.location.hash = anchor.hash
+      return
+    }
+  }
+}
+
+function gatherHeaderAnchors () {
+  const sidebarLinks = Array.from(document.querySelectorAll('.sidebar-group-items a.sidebar-link'))
```

Comment:
Use array spread instead of `Array.from`, since buble doesn't polyfill for that in IE11.

Reply:
I tried spread initially, but it seems to have some problems with NodeLists (https://gitlab.com/Rich-Harris/buble/issues/81). I would really prefer to use it here instead though. Do you have a preference for something else, assuming spread wouldn't work?